usaspending_api/etl/management/commands/swap_in_new_table.py

```diff
@@ -294,6 +306,43 @@ def validate_indexes(self, cursor):
                 )
                 raise SystemExit(1)
 
+    def dependent_views(self, cursor):
+        """ Detects views that are dependent on the table to be swapped. """
+        detect_dep_view_sql = f"""
+            SELECT
+                dependent_ns.nspname AS dep_view_schema,
+                dependent_view.relname AS dep_view_name,
+                CONCAT(dependent_ns.nspname, '.', dependent_view.relname) as dep_view_fullname,
+                RTRIM(pg_get_viewdef(CONCAT(dependent_ns.nspname, '.', dependent_view.relname), TRUE), ';') AS dep_view_sql
+            FROM pg_depend
+            JOIN pg_rewrite ON pg_depend.objid = pg_rewrite.oid
+            JOIN pg_class as dependent_view ON pg_rewrite.ev_class = dependent_view.oid
+            JOIN pg_class as source_table ON pg_depend.refobjid = source_table.oid
+            JOIN pg_attribute ON pg_depend.refobjid = pg_attribute.attrelid
+                AND pg_depend.refobjsubid = pg_attribute.attnum
+            JOIN pg_namespace dependent_ns ON dependent_ns.oid = dependent_view.relnamespace
+            JOIN pg_namespace source_ns ON source_ns.oid = source_table.relnamespace
+            WHERE
+                source_ns.nspname = '{self.curr_schema_name}'
+                AND source_table.relname = '{self.curr_table_name}'
+            GROUP BY
+                CONCAT(source_ns.nspname, '.', source_table.relname),
+                dependent_ns.nspname,
+                dependent_view.relname,
+                CONCAT(dependent_ns.nspname, '.', dependent_view.relname);
+        """
+        cursor.execute(detect_dep_view_sql)
+        dep_views = ordered_dictionary_fetcher(cursor)
+        return dep_views
+
+    def create_new_views(self, cursor):
+        for dep_view in self.dep_views:
+            # Note: Despite pointing at the new temp table at first,
+            #       the recreated views will automatically update and stay pointed to it in the swap via postgres
+            logger.info(f"Recreating dependent view: {dep_view['dep_view_fullname']}_temp")
+            dep_view_sql = dep_view["dep_view_sql"].replace(self.curr_table_name, self.temp_table_name)
+            cursor.execute(f"CREATE VIEW {dep_view['dep_view_fullname']}_temp AS ({dep_view_sql});")
```

Comment:
Would it be possible to create the temp view in the `temp` schema? I think this would prevent the `api_user` DB user from accessing the view. The schema would need to be updated to swapped in `swap_table_view_sql` as well

Reply:
It can be done. Right now it's keeping the temp views in the same schema as their main views, so if the user has access to the main view, it would have access to the temp view as well. Visa versa if they don't have access to the main view in the first place. I personally like it this way as the script is already agnostic to the `temp` schema, but I can hardcode it in if that's what we want.